unity/Assets/Scripts/IK_Robot_Arm_Controller.cs

```diff
@@ -517,17 +517,35 @@ public void DropObject()
             jointMeta.rootRelativePosition = FirstJoint.InverseTransformPoint(joint.position);
 
             //local rotation currently relative to immediate parent joint
-            joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
-            jointMeta.localRotation = new Vector4(vectorRot.x, vectorRot.y, vectorRot.z, angleRot);
+            if (joint.rotation != joint.GetChild(0).rotation)
+            {
+                joint.GetChild(0).localRotation.ToAngleAxis(out angleRot, out vectorRot);//getchild to grab the angler since that is what actually changes the geometry angle
```

Comment:
Just to be safe, can we add named arguments here, as in:

```c#
joint.GetChild(0).localRotation.ToAngleAxis(angle: out angleRot, axis: out vectorRot);
```
(note: angle and axis come from https://docs.unity3d.com/ScriptReference/Quaternion.ToAngleAxis.html)

Reply:
Sure. I extended that to all uses of ToAngleAxis, for consistency.